src/pip/_internal/operations/prepare.py

```diff
@@ -118,35 +118,6 @@ def _copy2_ignoring_special_files(src: str, dest: str) -> None:
         )
 
 
-def _copy_source_tree(source: str, target: str) -> None:
-    target_abspath = os.path.abspath(target)
-    target_basename = os.path.basename(target_abspath)
-    target_dirname = os.path.dirname(target_abspath)
-
-    def ignore(d: str, names: List[str]) -> List[str]:
-        skipped: List[str] = []
-        if d == source:
-            # Pulling in those directories can potentially be very slow,
-            # exclude the following directories if they appear in the top
-            # level dir (and only it).
-            # See discussion at https://github.com/pypa/pip/pull/6770
-            skipped += [".tox", ".nox"]
-        if os.path.abspath(d) == target_dirname:
-            # Prevent an infinite recursion if the target is in source.
-            # This can happen when TMPDIR is set to ${PWD}/...
-            # and we copy PWD to TMPDIR.
-            skipped += [target_basename]
-        return skipped
-
-    shutil.copytree(
-        source,
-        target,
-        ignore=ignore,
-        symlinks=True,
-        copy_function=_copy2_ignoring_special_files,
```

Comment:
@pradyunsg you can also remove `_copy2_ignoring_special_files` and the whole call tree from there (`copy2_fixed`, `is_socket` and related tests).

Reply:
Done!